hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractInvocationFuture.java

```diff
@@ -427,4 +430,17 @@ public String toString() {
             this.executor = executor;
         }
     }
+
+    @SuppressWarnings("checkstyle:visibilitymodifier")
+    protected static final class AltResult {
+        public final Throwable cause;
```

Comment:
Can we please make this field private and use a getter method to access it?

If we ever need to access the cause in a compatibility test, those field accesses are super painful (only method calls are properly proxied). I've been there with `Node` and all its public fields. I already see such a field access in `AbstractInvocationFuture_AbstractTest`.